spec/index.md

```diff
@@ -643,14 +663,56 @@ The HTTP response to this request can have the following responses:
+This request type can *also* be used to call a query method (but not a composite query method). A user may choose to go this way, instead of via the faster and cheaper [Request: Query call](#http-query) below, if they want to get a *certified* response. Note that the canister state will not be changed by sending an update call request type for a query method (except for cycle balance change due to message execution).
 
 :::note
 
 The functionality exposed via the [The IC management canister](#ic-management-canister) can be used this way.
 
 :::
 
+### Request: Sync Call {#http-sync-call}
+
+In order to make an update call to a canister and potentially get a synchronous response, the user makes a POST request to `/api/v2/canister/<effective_canister_id>/sync_call`. The request body consists of an authentication envelope with a `content` map with the following fields:
+
+-   `request_type` (`text`): Always `sync_call`
+
+-   `sender`, `nonce`, `ingress_expiry`: See [Authentication](#authentication)
+
+-   `canister_id` (`blob`): The principal of the canister to call.
+
+-   `method_name` (`text`): Name of the canister method to call.
+
+-   `arg` (`blob`): Argument to pass to the canister method.
+
+The HTTP response to this request can have the following responses:
+
+-   200 HTTP status with a non-empty body. The response contains the canister response, which is either a `reply` or a `reject`.
+    
+    -   If the update call resulted in a (replicated) reply, the response is a CBOR (see [CBOR](#cbor)) map with the following fields:
+
+        -   `status` (`text`): `"replied"`
+
+        -   `reply` (`blob`):  A certificate (see [Certification](#certification)) with subtrees at `/request_status/<request_id>` and `/time`. See [Request status](#state-tree-request-status) for more details on the request status.
+
+    -   If the update call resulted in a (non-replicated) reject, the response is a CBOR map with the following fields:
+
+        -   `status` (`text`): `"rejected"`
+
+        -   `reject_code` (`nat`): The reject code (see [Reject codes](#reject-codes)).
+
+        -   `reject_message` (`text`): a textual diagnostic message.
+
+        -   `error_code` (`text`): an optional implementation-specific textual error code (see [Error codes](#error-codes)).
+
+-   202 HTTP status with an empty body. This status is returned as a fallback mechanism for when a response from the canister was not produced in reasonable time. This implies that the request was accepted by the IC for further processing. Users should use [`read_state`](#http-read-state) to determine the status of the call.
```

Comment:
What will be a "reasonable time "? Should we just use the `ingress_expiry` value. The annoying part is that when the ingress_expiry is hit then we will be returning 202, which I am not sure how intuitive it is.

Reply:
Just my two cents here: AFAIK `ingress_expiry` is usually set to 5 minutes in the future. Connection timeouts for Web Applications happen usually around ~30 seconds (in my experience). So the reasonable time should be well below the 30 seconds mark, otherwise clients will start seeing HTTP 408 / HTTP 504 and the like (instead of the eventually produced 202 status).